src/config-utils.ts

```diff
@@ -60,6 +60,14 @@ const OVERLAY_MINIMUM_AVAILABLE_DISK_SPACE_MB = 20000;
 const OVERLAY_MINIMUM_AVAILABLE_DISK_SPACE_BYTES =
   OVERLAY_MINIMUM_AVAILABLE_DISK_SPACE_MB * 1_000_000;
 
+/**
+ * The minimum memory (in MB) that must be available for CodeQL to perform overlay
+ * analysis. If CodeQL will be given less memory than this threshold, then the
+ * action will not perform overlay analysis unless overlay analysis has been
+ * explicitly enabled via environment variable.
+ */
+const OVERLAY_MINIMUM_MEMORY_MB = 5 * 1024;
+
```

Comment:
Minor:

```suggestion
/**
 * The minimum memory (in MB) that must be available for CodeQL to perform overlay
 * analysis. If CodeQL will be given less memory than this threshold, then the
 * action will not enable overlay analysis unless overlay analysis has been
 * explicitly requested via environment variable.
 */
const OVERLAY_MINIMUM_MEMORY_MB = 5 * 1024;
```

More generally, I'd maybe avoid this level of detail because it is difficult to maintain and keep synchronised with code changes elsewhere. Perhaps something like:

```suggestion
/**
 * The minimum memory (in MB) that must be available for us to automatically
 * enable overlay analysis.
 */
const OVERLAY_MINIMUM_MEMORY_MB = 5 * 1024;
```